### DISCUSSION TITLE
Extend example with Apache TomEE application server

### DISCUSSION THREAD

**Discussion by rsoika** [imixs/imixs-jsf-example#61] [General]
Provide a docker based image to run the jsf-example app in [TomEE](http://tomee.apache.org/).

* provide Dockerfile for tomee
* provide tomee specific deployment descriptors (if needed)
* extend documentation

**Comment by 4asifm** [comment]
Startup of imixs-jsf-example-4.0.12 tested on the following environment

1.  OS - Windows ( 7 SP1) 
2. JDK - 1.8.0_121
3. Tomee plus - 8.0.0.M1 (see Note 1 for tomee-plus vs tomee-plume)
4. Mysql version 5.7.20 (CE)
5. created database imixs
6.  changed persistence details in persistence.xml in source of imixs-jsf-example-4.0.12  
7. compiled imixs-jsf-example-4.0.12 from source
8. created role `<role rolename="IMIXS-WORKFLOW-Manager"/>` - in tomcat-users.xml (conf folder)
9. created user named "both" `<user username="both" password="xxxxxx" roles="tomcat,role1,org.imixs.ACCESSLEVEL.MANAGERACCESS">` - in tomcat-users.xml
10. dropped eclipselink-2.4.2.jar in lib folder of tomee-plus 
11. dropped imixs-jsf-example-4.0.12.war file in webapps folder of tomee-plus
12. started tomee-plus startup script 
13. console error about missing imixs core libs in folder WEB-INF/classes for imixs-jsf-example-4.0.12 webapp
14. stopped the server and copied entire lib folder from WEB-INF to to WEB-INF/classes folder.
15. restarted tomee-plus
16. navigated to `http://localhost:<port-no>/imixs-jsf-example-4.0.12`
17. logged in as user "both" and password
18 . logging success - initial screen with "Task List" is visible.

Note 1: Same steps with tomee-plume didn't work. Log in worked, but there was no display of initial "Task List" table. As far as I could make out, tomee-plume has EclipseLink and Mojarra libraries in addition to all the libraries that tomee-plus has.

**Comment by rsoika** [comment]
That's great! I will extend the documentation

**Comment by rsoika** [comment]
I have now added your installation guide to the project home page:
https://imixs.org/doc/deployment/tomee.html
If you have any suggestions please let me know. Thanks for your help

**Comment by 4asifm** [comment]
Thanks.

I will keep an eye on the forum and see if i can answer any questions on
the topic

I will also post my experience about further development on the imixs
platform.

Kind Regards,
Asif

**Comment by 95pratik** [comment]
sir after successful deployment of imixs-jsf-example 4.2.0. how can we connect this application through our mysql database??

**Comment by rsoika** [comment]
What is the your use case? Do you want to write/read data from another table?
Take a look at the [Plugin API](https://www.imixs.org/doc/engine/plugins/index.html). With a custom plugin you can easily connect your workflow with other systems. Are you familiar with JPA?

**Comment by 95pratik** [comment]
sir I am trying to connect this application with mysql database so that whatever the ticket is generating on ui should be in our mysql db...

**Comment by rsoika** [comment]
Set your persistence-unit name to "org.imixs.workflow.jpa" not "imixs".

**Comment by beautifulnature** [comment]
I followed the above procedure in tomee plus. But receiving the following error
javax.el.ELException: org.imixs.workflow.exceptions.ModelException: Modelversion '1.1' not found! No matching version found for WorkflowGroup '' ($uniqueid=)

**Comment by rsoika** [comment]
This sounds like you have not uploaded the model. 

use this curl command:

    curl --user admin:adminpassword --request POST -Tsrc/workflow/ticket.bpmn http://localhost:8080/api/model/bpmn

You can verify the uploaded models in your browser:

http://localhost:8080/api/model

**Comment by beautifulnature** [comment]
please suggest windows equivalent command

**Comment by beautifulnature** [comment]
gr8 thankyou.  Its working. I installed curl.

**Comment by rsoika** [comment]
Thanks for your feedback, I updated the documentation.
